bootstrap/src/main/java/com/splunk/opentelemetry/javaagent/bootstrap/nocode/NocodeRules.java

```diff
@@ -29,23 +29,23 @@ public static final class Rule {
     private static final AtomicInteger counter = new AtomicInteger();
 
     private final int id = counter.incrementAndGet();
-    private final String className;
-    private final String methodName;
+    private final Object classMatcher; // ElementMatcher
```

Comment:
The reason we need this class in boot loader is for it to be usable from the advice. `NocodeInstrumentation` that uses the matchers has access to agent code. Perhaps it would help if we somehow took advantage of it, we could split the rule into stuff that is needed from the advice and from the agent code.

Reply:
Not sure I follow the idea here?  Are you suggesting something like a`bootstrap.NocodeRules` and `instrumentation.NocodeRules` with the same index used for both?

Reply:
What if you make `NocodeRules.Rule` an interface that includes only the methods needed by the advice. In the non-advice code you could use `RuleImpl` that implements `Rule` and provides more methods.